### PR TITLE
Post Settings - Tags

### LINKED ISSUE #5808: Post Settings: New screen to edit tags
Similar to https://github.com/wordpress-mobile/WordPress-Android/issues/5807, instead of editing the tags inline in the Post Settings screen. We'll display them inlined but we'll allow edition in a new screen (Title, textarea with tags suggestion, hint).

tags

### PR DESCRIPTION
**Pull request by oguzkocer**
Fixes #5808. This PR adds a new tags screen. Similar to excerpt and slug PRs, it doesn't handle the design changes inside the settings page. There might be some improvements to make in `EditPostSettingsFragment` but I am specifically staying away from making too many changes there for now. Once we have the other components, I am planning on a full review of that fragment to make the necessary improvements all together.

To test:
* Go into tags screen in post settings and make sure it's filtered correctly as you type.
* Selecting a tag should add it at the end.
* Duplicate tags are not handled for now. 
* Going back should update the tags text view in settings and updating the post should update the remote tags.

Just a reminder that, there is a v2 issue for tags #5921 and if there are any improvement ideas outside of the scope of this PR, we should consider adding them into that issue.

_Known issue:_ There is a jump when you go back from the tags page, I'll fix this when I work on the settings fragment.

screenshot_1495652580

### CODE REVIEW

**Comment by tonyr59h** [comment]
> Duplicate tags are not handled for now.

Is this tracked in an issue somewhere?

**Comment by tonyr59h** [comment]
[Long tags](https://cldup.com/FB31z8jbCG.png) should probably be single line and ellipsized or the second line will bleed into the view.

**Comment by tonyr59h** [comment]
Besides that single line issue things look great! Good work on this, love the upcoming v2 changes as well 😃

**Comment by oguzkocer** [comment]
> Is this tracked in an issue somewhere?

Duplicate tags handling tracked here: https://github.com/wordpress-mobile/WordPress-Android/issues/5921#issuecomment-305570758

**Comment by oguzkocer** [comment]
> Long tags should probably be single line and ellipsized or the second line will bleed into the view.

Thanks for reporting that @tonyr59h, I really didn't think of testing that. I think it could have been a fair solution to just let the height get bigger than the `40dp` we assigned, but for now I just ellipsized it in ee8e13a62f2986c8dac7590ab6a3ec7d96c00ca1 as you suggested. We can re-visit this in the v2 if necessary, it's not really a common use-case anyway. 

Should be ready for a second pass!

File: WordPress/src/main/java/org/wordpress/android/ui/posts/PostSettingsTagsActivity.java [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+    }
+
+    // Find the text after the last occurrence of "," and filter with it
+    private void filterListForCurrentText() {
+        String text = mTagsEditText.getText().toString();
+        int endIndex = text.lastIndexOf(",");
+        if (endIndex == -1) {
+            mAdapter.filter(text);
+        } else {
+            String textToFilter = text.substring(endIndex + 1);
```
Comment (maxme) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/PostSettingsTagsActivity.java]: I think we should trim `textToFilter` here, would work better if the user adds a whitespace after the comma.

Comment (oguzkocer) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/PostSettingsTagsActivity.java]: I was planning on asking @mattmiklic about that for the v2 as @tonyr59h already mentioned it https://github.com/wordpress-mobile/WordPress-Android/issues/5921#issuecomment-305067327. I made the change in 99da4042623a7814ca70cf1fef73d10398c808cf, however it feels a little weird if the space is added after some text and the suggestions stay the same.

**Comment by tonyr59h** [comment]
Looks good @oguzkocer! I noticed another styling issue but I'm going to add a comment to fix it up in v2 so we can get this shipped :shipit:

**Pull request merged**